src/user/Idealized_Hurricane.F90

```diff
@@ -325,16 +325,20 @@ subroutine idealized_hurricane_wind_forcing(sfc_state, forces, day, G, US, CS)
   enddo
 
   !> Get Ustar
-  do j=js,je
-    do i=is,ie
-      !  This expression can be changed if desired, but need not be.
-      forces%ustar(i,j) = G%mask2dT(i,j) * sqrt(US%L_to_Z * (CS%gustiness/CS%Rho0 + &
-              sqrt(0.5*(forces%taux(I-1,j)**2 + forces%taux(I,j)**2) + &
-                   0.5*(forces%tauy(i,J-1)**2 + forces%tauy(i,J)**2))/CS%Rho0))
-    enddo
-  enddo
+  if (associated(forces%ustar)) then ; do j=js,je ; do i=is,ie
+    !  This expression can be changed if desired, but need not be.
+    forces%ustar(i,j) = G%mask2dT(i,j) * sqrt(US%L_to_Z * (CS%gustiness/CS%Rho0 + &
+            sqrt(0.5*(forces%taux(I-1,j)**2 + forces%taux(I,j)**2) + &
+                 0.5*(forces%tauy(i,J-1)**2 + forces%tauy(i,J)**2))/CS%Rho0))
+  enddo ; enddo ; endif
+
+  !> Get tau_mag [R L Z T-2 ~> Pa]
+  if (associated(forces%tau_mag)) then ; do j=js,je ; do i=is,ie
+    forces%tau_mag(i,j) = G%mask2dT(i,j) * (CS%gustiness + &
+            sqrt(0.5*(forces%taux(I-1,j)**2 + forces%taux(I,j)**2) + &
+                 0.5*(forces%tauy(i,J-1)**2 + forces%tauy(i,J)**2)))
+  enddo ; enddo ; endif
```

Comment:
I guess these two could be combined with a `if (ustar .or. tau_mag)` (similarly in L676-683), but I'm probably already thinking too hard about the Idealized Hurricane case.

Reply:
In the next commit that I will put in after this one is resolved, we will only allocate one of forces%ustar or forces%tau_mag, but not both, so combining these two loops will become counterproductive, but at until that follow-on PR is added, this will not be obvious!